packages/edit-post/src/index.js

```diff
@@ -79,22 +79,32 @@ export function initializeEditor(
 	}
 
 	/*
-	 * Prevent adding template part and post content block in the post editor.
+	 * Prevent adding template part and post content (except in query block) in the post editor.
 	 * Only add the filter when the post editor is initialized, not imported.
 	 * Also only add the filter(s) after registerCoreBlocks()
 	 * so that common filters in the block library are not overwritten.
 	 */
 	addFilter(
 		'blockEditor.__unstableCanInsertBlockType',
-		'removeTemplatePartsFromInserter',
-		( canInsert, blockType ) => {
-			if (
-				! select( editPostStore ).isEditingTemplate() &&
-				( blockType.name === 'core/template-part' ||
-					blockType.name === 'core/post-content' )
-			) {
+		'removeTemplatePartsAndPostContentFromInserter',
```

Comment:
In hindsight I wonder if it might be better to make a second call to `addFilter` for the post content block and leave the template part code in tact.

The reason is that some devs might use `removeFilter( 'blockEditor.__unstableCanInsertBlockType', 'removeTemplatePartsFromInserter' )`, if the name is changed it would stop that `removeFilter` call from working.

Reply:
😄  I approved before seeing your comment. This makes sense.

Reply:
Oh ya, good thought.

Reply:
Good spotting Dan, thanks, will update.